karma.conf.js

```diff
@@ -4,18 +4,30 @@ if (!process.env.TRAVIS){
     process.env.CHROME_BIN = require('puppeteer').executablePath();
 }
 
+function getTestFiles(){
+    switch (process.env.BROWSER_BUNDLE_TEST){
+        case 'publishedDist': return ["packages/web3/dist/web3.min.js", "test/e2e.minified.js"]
+        case 'gitRepoDist':   return ["dist/web3.min.js", "test/e2e.minified.js"]
+        default:              return ["test/**/e2e*.js"]
+    }
+}
+
+// Only loads preprocessor for the logic unit tests
+function getPreprocessors(){
+    if (!process.env.BROWSER_BUNDLE_TEST){
```

Comment:
We use more and more such environment vars and I think it would be the time now or at least any time soon to document those environment variables and to define the naming rule behind it. It is already hard to follow the different execution paths depending on the defined environment variables. :)

Reply:
Yes, I wonder where should we define them and how improve their clarity though. I will have to think about this a bit - probably a separate PR because we have 5 or 6 of these now.

Reply:
I only know [convict](https://github.com/mozilla/node-convict) which could improve the situation here (I've never used it until now).